### ISSUE TITLE
Example showing errors

### ISSUE THREAD

**Issue by domgolonka** [patrikeh/go-deep#22]
Hi, the example in the readme is giving errors:

// params: learning rate, momentum, alpha decay, nesterov
optimizer := training.NewSGD(0.05, 0.1, 1e-6, true)
// params: optimizer, verbosity (print stats at every 50th iteration)
trainer := training.NewTrainer(optimizer, 50)

Cannot use optimizer as (type *SGD) as type Solver

**Comment by patrikeh** [comment]
Not sure how that happened but the example runs fine for me:

[https://play.golang.org/p/8_anucxejrS](https://play.golang.org/p/8_anucxejrS)
